cirq/google/engine/calibration.py

```diff
@@ -103,8 +113,44 @@ def __len__(self) -> int:
         return len(self._metric_dict)
 
     def __str__(self) -> str:
-
-        return 'Calibration(keys={})'.format(list(sorted(self.keys())))
+        return f'Calibration(keys={list(sorted(self.keys()))})'
+
+    def __repr__(self) -> str:
+        return ('cirq.google.Calibration(metrics='
+                f'{repr(dict(self._metric_dict))})')
+
+    def to_proto(self) -> v2.metrics_pb2.MetricsSnapshot:
+        """Reconstruct the protobuf message represented by this class."""
+        proto = v2.metrics_pb2.MetricsSnapshot()
+        for key in self._metric_dict:
+            for target, value_list in self._metric_dict[key].items():
+                current_metric = proto.metrics.add()
+                current_metric.name = key
+                current_metric.targets.extend(
+                    [v2.qubit_to_proto_id(q) for q in target])
+                for value in value_list:
+                    current_value = current_metric.values.add()
+                    if isinstance(value, float):
+                        current_value.double_val = value
+                    if isinstance(value, int):
+                        current_value.int64_val = value
+                    if isinstance(value, str):
```

Comment:
Should this cause an error if there's a bad type rather than failing silently?

Reply:
Done.